.github/workflows/release.yml

```diff
@@ -37,11 +37,17 @@ jobs:
         run: npm ci
         shell: bash
 
-      - name: Release and publish
-        run: npx nx release version ${{ github.event.release.tag_name }} --yes
+      - name: Release new version
```

Comment:
Not sure I understand the difference between `release version` and `release publish`? Is `release version` just incrementing the version number?

Reply:
You're correct.
- `nx release version` only bump version number in `package.json` to the version number in the `tag_name`.
- `nx release publish` will actually publish the package to npm.

You can read more about `nx release` from here: https://nx.dev/features/manage-releases

Reply:
These are in two different steps though, so the version number change in package.json won't be carried across

Reply:
The last step in this workflow is actually a git push. Will that help to carry over the version number change?

```
- name: Push version changes
  run: git push
  env:
    github-token: ${{ secrets.GITHUB_TOKEN }}
```

Reply:
A change to `package.json` should carry over between steps shouldn't it? Steps in a workflow operate on the same files. Otherwise the standard install step followed by some other step wouldn't work.

Having said that @kai-nguyen-aligent is there a reason you've picked out `nx release version` and `nx release publish` rather than just using the whole `nx release` process?

Reply:
Sorry brain-fart moment, steps do carry changes across, builds don't carry changes :sweat_smile: 

But yes, I am also curious why not just the one step?

Reply:
If we use the whole `nx release`, we'll need to add the configuration into each `project.json` file for each package. I feel like it's more consistent and clearer when we do this in pipeline only.

Reply:
As long as it functionally does what we want I'm happy. Probably a good idea to document these weird edges somewhere though.